shipyard-controller/handler/shipyardcontroller.go

```diff
@@ -102,6 +103,22 @@ func (sc *shipyardController) HandleIncomingEvent(event models.Event) error {
 				event.Triggeredid,
 			); err != nil {
 				sc.logger.Error(fmt.Sprintf("could not update event for project %s: %s", eventData.Project, err.Error()))
+				return
+			}
+
+			// also update the shipyard in the materialized view because it might have been changed in the meantime
+			shipyard, err := common.GetShipyard(eventData)
+			if err != nil {
+				sc.logger.Error(fmt.Sprintf("could not retrieve shipyard file of project %s: %s", eventData.Project, err.Error()))
+				return
+			}
+			shipyardContent, err := yaml.Marshal(shipyard)
+			if err != nil {
+				sc.logger.Error(fmt.Sprintf("could not encode shipyard file of project %s: %s", eventData.Project, err.Error()))
+				return
+			}
+			if err := sc.eventsDbOperations.UpdateShipyard(eventData.Project, string(shipyardContent)); err != nil {
+				sc.logger.Error(fmt.Sprintf("could not update shipyard content of project %s: %s", eventData.Project, err.Error()))
```

Comment:
I think having this in `HandleIncomingEvent` is suboptimal, as we don't need to retrieve the shipyard for every event we process (in fact, we might not even want that in the future).

Looking at shipyardcontroller.go, I think a better place would be to have it in `handleTriggeredEvent`, where we already retrieve the shipyard.

Reply:
agreed, have moved it to `handleTriggeredEvent`